normandy/recipes/api/views.py

```diff
@@ -88,14 +89,48 @@ def update(self, request, *args, **kwargs):
         Intercept PUT requests and have them create instead of update
         if the object does not exist.
         """
-        if request.method == 'PUT':
+        if request.method in ['PUT', 'PATCH']:
             try:
-                self.get_object()
+                recipe = self.get_object()
             except Http404:
-                return self.create(request, *args, **kwargs)
+                if request.method == 'PUT':
+                    return self.create(request, *args, **kwargs)
+            else:
+                if recipe.is_approved:
+                    recipe.disable(ignore_revision_id=True)
```

Comment:
Ah, I missed that. It isn't totally clear from the method name. I was imagining an approved but disabled recipe (though since any change rightfully unapproves, that recipe wouldn't be very useful).